File: common/src/main/java/org/sandboxpowered/silica/network/play/serverbound/PlayerRotation.java
package org.sandboxpowered.silica.network.play.serverbound;

import org.sandboxpowered.silica.network.*;

public class PlayerRotation implements PacketPlay {
    private float yaw;
    private float pitch;
    private boolean onGround;

    public PlayerRotation() {
    }

    public PlayerRotation(float yaw, float pitch, boolean onGround) {
        this.yaw = yaw;
        this.pitch = pitch;
        this.onGround = onGround;
    }

    @Override
    public void read(PacketByteBuf buf) {
        yaw = buf.readFloat();
        pitch = buf.readFloat();
        onGround = buf.readBoolean();
    }

    @Override
    public void write(PacketByteBuf buf) {
        buf.writeFloat(yaw);
        buf.writeFloat(pitch);
        buf.writeBoolean(onGround);
    }

    @Override
    public void handle(PacketHandler packetHandler, PlayContext context) {
        context.mutatePlayerJava(input -> {
            input.setWantedYaw(this.yaw);
            input.setWantedPitch(this.pitch);
        });
    }
}

File: common/src/main/kotlin/org/sandboxpowered/silica/network/play/serverbound/PlayerRotation.kt
package org.sandboxpowered.silica.network.play.serverbound

import org.sandboxpowered.silica.component.VanillaPlayerInput
import org.sandboxpowered.silica.network.PacketByteBuf
import org.sandboxpowered.silica.network.PacketHandler
import org.sandboxpowered.silica.network.PacketPlay
import org.sandboxpowered.silica.network.PlayContext

class PlayerRotation(
    private var yaw: Float = 0f,
    private var pitch: Float = 0f,
    private var onGround: Boolean = false,
) : PacketPlay {
    override fun read(buf: PacketByteBuf) {
        yaw = buf.readFloat()
        pitch = buf.readFloat()
        onGround = buf.readBoolean()
    }

    override fun write(buf: PacketByteBuf) {
        buf.writeFloat(yaw)
        buf.writeFloat(pitch)
        buf.writeBoolean(onGround)
    }

    override fun handle(packetHandler: PacketHandler, context: PlayContext) {
        context.mutatePlayer {
            it.wantedYaw = yaw
            it.wantedPitch = pitch
        }
    }
}